FacilityBooking/FacilitiesAPI.py

```diff
@@ -0,0 +1,203 @@
+from db import *
+from flask import Flask, render_template, flash, redirect, url_for, request, jsonify, make_response
+import os
+from datetime import datetime
+import time
+from flask_cors import cross_origin
+from flask import Blueprint
+import pymongo
+import sys
+sys.path.append("../db")
+
+facilities_api = Blueprint("facilities", __name__)
+
+
+@facilities_api.route('/')
+@cross_origin()
+def root_route():
+    return 'What up losers'
+
+
+@facilities_api.route('/facilities', methods=["GET"])
+@cross_origin(supports_credentials=True)
+def all_facilities():
+    try:
+        data = list(db.Facilities.find(
+            {}, {"_id": 0}).sort([("facilityName", 1)]))
+        response = {"status": "success", "data": {"facilities": data}}
+    except Exception as e:
+        print(e)
+        return {"err": str(e), "status": "failed"}, 400
+    return make_response(response)
+
+
+@ facilities_api.route('/facilities/<facilityID>', methods=["GET"])
+@ cross_origin(supports_credentials=True)
+def get_facility_name(facilityID):
+    try:
+        data = list(db.Facilities.find(
+            {"facilityID": int(facilityID)}, {"_id": 0}))
+        if len(data) == 0:
+            raise Exception("Facility not found")
+        response = {"status": "success", "data": data}
+    except Exception as e:
+        return {"err": str(e), "status": "failed"}, 400
+    return make_response(response)
+
+
+@ facilities_api.route('/bookings/<bookingID>', methods=["GET"])
+@ cross_origin(supports_credentials=True)
+def get_one_booking(bookingID):
+    try:
+        data = list(db.Bookings.find(
+            {"bookingID": int(bookingID)}, {"_id": 0}))
+        response = {"status": "success", "data": data}
+        if len(data) == 0:
+            raise Exception("Booking not found")
+    except Exception as e:
+        return {"err": str(e), "status": "failed"}, 400
+    return make_response(response)
+
+
+@ facilities_api.route('/bookings/user/<userID>', methods=["GET"])
+@ cross_origin(supports_credentials=True)
+def user_bookings(userID):
+    try:
+        data = list(db.Bookings.find({"userID": userID}, {
+                    "_id": 0}).sort([("startTime", 1)]))
+        response = {"status": "success", "data": data}
+    except Exception as e:
+        return {"err": str(e), "status": "failed"}, 400
```

Comment:
Just curious, does exceptions like these stop the app? or only unhandled ones do? either way, 
![image](https://user-images.githubusercontent.com/26619858/113256198-6e4e3d80-92fb-11eb-821d-7668c5f50475.png)

Reply:
Only unhandled exceptions stop the app